File: mobile/android/DriverBehaviorModel.java
package com.autoconnect.android;

/** Driver behaviour metrics and derived score. */
public class DriverBehaviorModel {
    public int hardBrakes;
    public int rapidAccelerations;
    public int overspeedEvents;
    public int driverScore;      // 0-100
    public String classification; // SAFE | MODERATE | RISKY

    public DriverBehaviorModel(int hardBrakes, int rapidAccelerations,
                               int overspeedEvents, int driverScore,
                               String classification) {
        this.hardBrakes = hardBrakes;
        this.rapidAccelerations = rapidAccelerations;
        this.overspeedEvents = overspeedEvents;
        this.driverScore = driverScore;
        this.classification = classification;
    }
}

File: mobile/ios/DriverBehaviorModel.swift
import Foundation

/// Driver behaviour metrics and derived score.
struct DriverBehaviorModel: Codable {
    let hardBrakes: Int
    let rapidAccelerations: Int
    let overspeedEvents: Int
    let driverScore: Int
    let classification: String  // SAFE | MODERATE | RISKY

    enum CodingKeys: String, CodingKey {
        case hardBrakes = "hard_brakes"
        case rapidAccelerations = "rapid_accelerations"
        case overspeedEvents = "overspeed_events"
        case driverScore = "driver_score"
        case classification
    }
}
